File: Course-Registration-System-main/java-(iteration_1)/src/Student.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Student extends User  {
    private int semester;
    private String major;
    private int advisorID;
    private Advisor advisor;
    private ArrayList<CourseSection> currentCourses;
    private Transcript transcript;
    private ArrayList<CourseRequest> courseRequests;

    // Constructor

    public Student(int id, String password, String name, String surname, int semester, String major, int advisorID, Advisor advisor
                   , ArrayList<CourseSection> currentCourses, Transcript transcript, ArrayList<CourseRequest> courseRequests) {
        super(id, password, name, surname);
        this.semester = semester;
        this.major = major;
        this.advisorID = advisorID;
        this.advisor = advisor;
        this.currentCourses = currentCourses;
        this.transcript = transcript;
        this.courseRequests = courseRequests;
    }

    // Getter & Setter methods

    public int getSemester() {
        return semester;
    }

    public void setSemester(int semester) {
        this.semester = semester;
    }

    public String getMajor() {
        return major;
    }

    public void setMajor(String major) {
        this.major = major;
    }

    public Advisor getAdvisor() {
        return advisor;
    }

    public void setAdvisor(Advisor advisor) {
        this.advisor = advisor;
    }
    public int getAdvisorID() {
        return advisorID;
    }
    public void setAdvisorID(int advisorID) {
        this.advisorID = advisorID;
    }
    public ArrayList<CourseSection> getCurrentCourses() {
        return currentCourses;
    }

    public void setCurrentCourses(ArrayList<CourseSection> currentCourses) {
        this.currentCourses = currentCourses;
    }

    public Transcript getTranscript() {
        return transcript;
    }

    public void setTranscript(Transcript transcript) {
        this.transcript = transcript;
    }

    public ArrayList<CourseRequest> getCourseRequests() {
        return courseRequests;
    }

    public void setCourseRequests(ArrayList<CourseRequest> courseRequests) {
        this.courseRequests = courseRequests;
    }

    public void save() {
        // implementation
    }

    @Override
    public String getInfo() {
        // Implementation for viewing general student information
        return ("\nStudent ID: " + getId() + "\nFull Name: " + getName() + " " + getSurname() + "Major: " + getMajor()
                + "\nSemester: "
                + getSemester() + "\nMajor: " + getMajor() + "\n"  + getAdvisor().getInfo()
                + "\n--------------------------");
    }


    // Lists student's current courses
    public String[] listCurrentCourses() {
        String[] currentCourseList = new String[50];
        if (currentCourses != null) {
            for (int i = 0; i < currentCourses.size(); i++) {
                currentCourseList[i] = currentCourses.get(i).viewCourseSectionInfo();
            }
        }
        return currentCourseList;
    }
}

File: Course-Registration-System-main/python-(iteration_2)/src/Student.py
from User import User
from Advisor import Advisor
from CourseSection import CourseSection
from Transcript import Transcript
from CourseRequest import CourseRequest
from StudentSaver import StudentSaver

class Student(User):
    def __init__(self, id, password, name, surname, semester, major, advisorID, advisor, currentCourses, transcript, courseRequests):
        super().__init__(id, password, name, surname)
        self.__semester = semester
        self.__major = major
        self.__advisorID = advisorID
        self.__advisor = advisor
        self.__currentCourses = currentCourses
        self.__transcript = transcript
        self.__courseRequests = courseRequests

    # Getter & Setter methods
    def getSemester(self):
        return self.__semester

    def setSemester(self, semester):
        self.__semester = semester

    def getMajor(self):
        return self.__major

    def setMajor(self, major):
        self.__major = major

    def getAdvisor(self):
        return self.__advisor

    def setAdvisor(self, advisor):
        self.__advisor = advisor

    def getAdvisorId(self):
        return self.__advisorID

    def setAdvisorId(self, advisorID):
        self.__advisorID = advisorID

    def getCurrentCourses(self):
        return self.__currentCourses

    def setCurrentCourses(self, currentCourses):
        self.__currentCourses = currentCourses

    def getTranscript(self):
        return self.__transcript

    def setTranscript(self, transcript):
        self.__transcript = transcript

    def getCourseRequests(self):
        return self.__courseRequests

    def setCourseRequests(self, courseRequests):
        self.__courseRequests = courseRequests

    def save(self):
        StudentSaver.save(self.get_id(), self.get_current_courses(), self.get_course_requests())
        pass

    def getInfo(self):
        # Implementation for viewing general student information
        return ("\nStudent ID: " + str(self.getId()) + "\nFull Name: " + self.getName() + " " + self.getSurname() +
                "\nMajor: " + self.getMajor() + "\nSemester: " + str(self.getSemester()) +
                "\n" + self.getAdvisor().getInfo() +
                "\n--------------------------")

    # Lists student's current courses
    def listCurrentCourses(self):
        current_course_list = []
        if self.__currentCourses is not None:
            for course_section in self.__currentCourses:
                current_course_list.append(course_section.viewCourseSectionInfo())
        return current_course_list
